enginetest/queries/index_queries.go

```diff
@@ -3764,6 +3764,253 @@ var IndexPrefixQueries = []ScriptTest{
 			},
 		},
 	},
+	{
+		// https://github.com/dolthub/dolt/issues/7040
+		Name: "unique indexes on TEXT/BLOB columns with no prefix length (MariaDB compatibility)",
+		SetUpScript: []string{
+			"create table t (pk int primary key, col1 text);",
+			"create table j2 (pk int primary key, col1 varchar(100));",
+			"insert into j2 values (1, '100'), (2, '  '), (3, '300');",
+		},
+		Assertions: []ScriptTestAssertion{
+			{
+				Query:    "select @@strict_mysql_compatibility;",
+				Expected: []sql.Row{{0}},
+			},
+			{
+				Query:    "alter table t add unique key k1(col1);",
+				Expected: []sql.Row{{types.NewOkResult(0)}},
+			},
+			{
+				Query:    "insert into t values (4, ''), (5, ' '), (8, NULL), (-1, '  ');",
+				Expected: []sql.Row{{types.NewOkResult(4)}},
+			},
+			{
+				Query:    "insert into t values (1, 'oneasdfasdf');",
+				Expected: []sql.Row{{types.NewOkResult(1)}},
+			},
+			{
+				Query:       "insert into t values (2, 'oneasdfasdf');",
+				ExpectedErr: sql.ErrUniqueKeyViolation,
+			},
+			{
+				// Skipped until Dolt's implementation can return this error message with the raw
+				// content value, and not the hashed value.
+				Skip:           true,
+				Query:          "insert into t values (2, 'oneasdfasdf');",
+				ExpectedErrStr: "duplicate unique key given: [oneasdfasdf]",
+			},
+			{
+				Query:              "select col1 from t where col1='oneasdfasdf';",
+				Expected:           []sql.Row{{"oneasdfasdf"}},
+				CheckIndexedAccess: true,
+				ExpectedIndexes:    []string{"k1"},
+			},
+			{
+				// Indexes with content-hashed fields are not eligible for use with range scans
+				Query: "explain select * from t where col1 >= 'one';",
+				Expected: []sql.Row{
+					{"Filter"},
```

Comment:
Yeah, my gut was telling me to pull out the plan tests since most had matching tests using `ScriptTest.ExpectedIndexes`. Cleaned up in 1885159ec0b52f1e798af67320385a477a333682.